docs/use-cases/microfrontends.md

```diff
@@ -0,0 +1,307 @@
+    rules:
+      - key: "1"
+        segments: "*"
+        percentage: 100
+
+  production:
+    rules:
+      - key: "1"
+        segments: "*"
+        percentage: 100
+```
+
+## Tagging features
+
+When we create or update any feature in your Featurevisor project, we can tag it with the microfrontend it belongs to:
+
+```yml
+# features/showMarketingBanner.yml
+description: Shows marketing banner at the bottom of the page
+tags:
+  - products # tagging with `products` microfrontend
+
+# ...
+```
+
+It is possible some features may overlap across microfrontends. For example, we may want to show a marketing banner to users in both the `products` and `checkout` microfrontends. In that case, we can tag the feature with both microfrontends:
+
+```yml
+# features/showMarketingBanner.yml
+description: Shows marketing banner at the bottom of the page
+tags:
+  - products
+  - checkout
+
+# ...
+```
+
+## Anonymous vs Authenticated
+
+Featurevisor relies on the feature's `bucketBy` property to determine how to [bucket](/docs/bucketing) the user into a variation.
+
+It's an easy decision to make when choosing the `bucketBy` value when the microfrontend is only accessible to either anonymous or authenticated users, and not both.
+
+| Microfrontend | Access          | `bucketBy` |
+|---------------|-----------------|------------|
+| `signup`      | Anonymous users | `deviceId` |
+| `signin`      | Anonymous users | `deviceId` |
+| `account`     | Authenticated   | `userId`   |
+
+But what shall we do for microfrontends that are accessible to both anonymous and authenticated users? Like `products` and `checkout` in our example.
+
+### Design decision
+
+It's a design decision that we need to take here when defining our features.
+
+We can either choose to:
+
+- always use `deviceId` attribute for `bucketBy`, irrespective of whether the user is anonymous or authenticated, or
+- we can choose to use `deviceId` for anonymous users and `userId` for authenticated users.
+
+The drawback of using `deviceId` at all times for both anonymous and authenticated users is that it will result in inconsistent variations for logged in users across multiple sessions or devices.
+
+If we wish to get the maximum benefit of Featurevisor's consistent bucketing making sure the same logged in user sees the same variation across all devices and sessions, we can use:
+
+- `userId` for authenticated users, and
+- `deviceId` for anonymous users
+
+### Bucket by first available attribute
+
+We can express that in our feature as follows:
+
+```yml
+# features/showMarketingBanner.yml
+description: Shows marketing banner at the bottom of the page
+tags:
+  - products
+  - checkout
+
+bucketBy:
+  # if `userId` is available then it will be used,
+  # otherwise it will fall back to `deviceId`
+  or:
+    - userId
+    - deviceId
+
+# ...
+```
+
+This will make sure when `userId` attribute's value is available, it will be used for bucketing. Otherwise, it will fall back to `deviceId`.
```

Comment:
Here I'd expect an explanation of what "value is available" means exactly in this context. For example, will the `undefined` and empty string be treated the same and be fell down to `deviceId`? 

In general, we might think of some way to set the availability rules on the attribute definition. With that you can be even more precise and work around cases where, for instance, application assign temporarily user ids in special format for unauthorised users.